controllers/operator/apigateway_controller.go

```diff
@@ -73,17 +82,21 @@ func (r *APIGatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	if kymaGatewayStatus := gateway.ReconcileKymaGateway(ctx, r.Client, &apiGatewayCR); !kymaGatewayStatus.IsReady() {
+	if kymaGatewayStatus := gateway.ReconcileKymaGateway(ctx, r.Client, &apiGatewayCR, APIGatewayResourceListDefaultPath); kymaGatewayStatus.IsError() || kymaGatewayStatus.IsWarning() {
 		return r.requeueReconciliation(ctx, apiGatewayCR, kymaGatewayStatus)
 	}
 
+	if apiGatewayStatus := r.apiGatewayReconciliation.Reconcile(ctx, &apiGatewayCR); apiGatewayStatus.IsError() || apiGatewayStatus.IsWarning() {
```

Comment:
I agree, just am not sure if this was breaking some test. If not mistaken the order has to be exactly like that. But let see, I will move it up and we can check the tests.

Reply:
Now I remember. If I move the more general condition first, I can not test for the more specific condition with the integration tests (because more general will fail first). I decided I like to have the tests for the more specific conditions instead of fail earlier. If you think failing early on any APIRule would be more important, we may also reconsider if we really need the check for APIRule referring to Kyma Gateway at all, since any APIRule check will basically overwrite it in this case, what do you think ?

Reply:
Ah I understand, but I think we can still check that the resource blocking is in place for Kyma Gateway, by testing it with Virtual Service blocking the deletion ( this test already exists). And then rely on unit testing for APIRule also blocking the deletion of Kyma Gateway. I think in this case it would be okay to not have an integration test for all possible blocking resources.
Additionally, I think we should keep the logic for blocking Gateway.deletion by APIRule-usage, since this will also be relevant once we allow customer created gateways managed by APIGateway CR. So for Kyma Gateway it might not be so relevant, but for future implementation of gateway management it will be. So the current concrete implementation will be made more generic.

Reply:
Make sense, will move CR finalizer reconciliation as first and adapt the tests.